fvtest/compilerunittest/CodeGenTest.hpp

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef CODEGENTEST_HPP
+#define CODEGENTEST_HPP
+
+#include <gtest/gtest.h>
+
+#include "Jit.hpp"
+#include "codegen/CodeGenerator.hpp"
+#include "compile/Compilation.hpp"
+#include "env/ConcreteFE.hpp"
+#include "env/SystemSegmentProvider.hpp"
+#include "ilgen/IlGenRequest.hpp"
+#include "ilgen/IlGeneratorMethodDetails.hpp"
+#include "ilgen/TypeDictionary.hpp"
+
+#include "codegen/CodeGenerator_inlines.hpp"
+#include "compile/Compilation_inlines.hpp"
+#include "ilgen/IlGeneratorMethodDetails_inlines.hpp"
+
+namespace TRTest {
+
+class NullIlGenRequest : public TR::IlGenRequest {
+    TR::IlGeneratorMethodDetails _details;
+
+public:
+    NullIlGenRequest() : TR::IlGenRequest(_details) {}
+
+    virtual TR_IlGenerator *getIlGenerator(
+        TR::ResolvedMethodSymbol *methodSymbol,
+        TR_FrontEnd *fe,
+        TR::Compilation *comp,
+        TR::SymbolReferenceTable *symRefTab
+    ) {
+        throw 0;
```

Comment:
This override is necessary since `TR::IlGenRequest` has this as a pure virtual function, even though it's doesn't seem to be used anywhere outside ilgen. As for the throw itself, that was a placeholder while I was verifying that assumption. I meant to replace it with a more descriptive exception type before submitting this code, although I seem to have forgotten to do that. Any thoughts on what should be done about this?

Reply:
I see.

I'd suggest using GTests `FAIL()` macro. Maybe something like:

```c++
        FAIL() << "Test suite expects this method to never be called.";
```

Reply:
Looks like the Googletest `FAIL()` macro can't be used here for some reason, as it causes a compilation error. I've switched to throwing an actual exception in 3c75b24c0c32f8aba54383c21b0b455a2f52ea7d.